### PR TITLE
[raz_s3] Do not enable S3 browser in ADLS

### PR DESCRIPTION
**Pull request by Harshg999**
## What changes were proposed in this pull request?

- Check for RAZ URL and AWS_ACCOUNT Hostname in the config this time.
- In RAZ ADLS cluster, the AWS_ACCOUNT Hostname should not be present practically.

## How was this patch tested?

- Successfully updating and running unit tests.

### CODE REVIEW

File: desktop/libs/aws/src/aws/conf.py [hunk 1]
```diff
@@ -349,13 +349,13 @@ def has_s3_access(user):
...
-    is_admin(user) or user.has_hue_permission(action="s3_access", app="filebrowser") or RAZ.IS_ENABLED.get())
+    is_admin(user) or user.has_hue_permission(action="s3_access", app="filebrowser") or is_raz_s3())
 
 
-def has_raz_s3():
+def is_raz_s3():
   from desktop.conf import RAZ  # Must be imported dynamically in order to have proper value
 
-  return (RAZ.IS_ENABLED.get() and bool(get_raz_s3_default_bucket()))
+  return (RAZ.IS_ENABLED.get() and AWS_ACCOUNTS['default'].HOST.get())
```
Comment (romainr) [inline desktop/libs/aws/src/aws/conf.py:358]: We would need to contain all the logic inside: e.g.

     IS_SELF_SIGNING_ENABLED.get() or \
`      ( and 'default' in list(AWS_ACCOUNTS.keys()) and AWS_ACCOUNTS['default'].get_raw() and AWS_ACCOUNTS['default'].HOST.get()`

File: desktop/libs/aws/src/aws/conf_tests.py [hunk 2]
```diff
@@ -66,9 +67,18 @@ def test_has_s3_access_when_raz_enabled(self):
...
+    resets = [
+      RAZ.IS_ENABLED.set_for_testing(True),
+      conf.AWS_ACCOUNTS.set_for_testing({'default': {
+        'region': 'us-west-2',
+        'host': 's3-us-west-2.amazonaws.com',
+        'allow_environment_credentials': 'false'
+      }})
+    ]
     try:
       assert_true(conf.has_s3_access(self.user))
```
Comment (romainr) [inline desktop/libs/aws/src/aws/conf_tests.py:80]: We we don't also test `is_raz_s3` too?

**Review by romainr — COMMENTED** [review]
Nice, almost!

File: desktop/libs/aws/src/aws/conf.py (hunk 1 above)
Comment (Harshg999) [reply desktop/libs/aws/src/aws/conf.py:358]: I see!

File: desktop/libs/aws/src/aws/conf_tests.py (hunk 2 above)
Comment (Harshg999) [reply desktop/libs/aws/src/aws/conf_tests.py:80]: Added

File: desktop/libs/aws/src/aws/conf.py [hunk 3]
```diff
@@ -288,7 +288,7 @@ def is_enabled():
       has_iam_metadata() or \
       conf_idbroker.is_idbroker_enabled('s3a') or \
       IS_SELF_SIGNING_ENABLED.get() or \
```
Comment (romainr) [inline desktop/libs/aws/src/aws/conf.py:290]: Can be removed now

File: desktop/libs/aws/src/aws/conf_tests.py [hunk 4]
```diff
@@ -66,9 +67,39 @@ def test_has_s3_access_when_raz_enabled(self):
...
+    assert_false(conf.is_raz_s3())
+
+    # When RAZ is enabled for S3
+    resets = [
+      RAZ.IS_ENABLED.set_for_testing(True),
+      conf.AWS_ACCOUNTS.set_for_testing({'default': {
+        'region': 'us-west-2',
+        'host': 's3-us-west-2.amazonaws.com',
+        'allow_environment_credentials': 'false'
+      }})
```
Comment (romainr) [inline desktop/libs/aws/src/aws/conf_tests.py:98]: PS: better test would have been to test first with Raz enabled only that we still are on False

**Review by romainr — APPROVED** [review]
:+1:

**Comment by Harshg999** [comment]
@romainr  Hum, the py3.8 build looks fishy in the CI checks and not compiling (after re-running). Maybe needs an update?

**Pull request merged**